charmcraft.yaml

```
@@ -43,3 +43,21 @@ parts:
       # Pydantic is not actually used by the lib but charmcraft install pydeps from libs even if
       # they are not used.
       - pydantic>2.0
+    plugin: charm
+    build-packages: [git]
+
+config:
+  options:
+    ready-timeout:
+      type: int
+      default: 100
+      description: >
+        The maximum time (in seconds) to wait for the waypoint deployment to be 
+        ready. This applies specifically to the deployment created for the Istio 
+        waypoint controller. If the deployment does not become ready within this time, 
+        charm will go into blocked state.
+    model-on-mesh:
+      type: boolean
+      default: false
+      description: >
+        Add the whole model on the mesh and connect it to the waypoint
```

Just a suggestion.
```suggestion
        Add this charm's model to the service mesh. 
 All charms in this model will automatically be added to the mesh. 

```

Not sure if you need to mention the waypoint here.  I think somewhere in readme or elsewhere we'd describe what this charm does (incl deploy a waypoint) and then this becomes redundant here